Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -281,7 +281,7 @@ class BrowserViewController: UIViewController {
         })
 
 
-        searchLoader = SearchLoader(history: profile.history, urlBar: urlBar)
+        searchLoader = SearchLoader(profile: profile, urlBar: urlBar)
```

Comment:
This isn't the first time we've had to tiptoe around subtle DB initialization bugs. These hidden side effects are too error-prone -- can we not create something more robust?

> The fix for that is to… not do that. SearchLoader can access profile.history when it needs it, on the correct thread.

Hm, what do you mean by "correct thread"? [Looks like `query` is still set on the UI thread.](https://github.com/mozilla/firefox-ios/blob/9a5eeccebda4f1594b95e1c174045dfe840a910c/Client/Frontend/Browser/BrowserViewController.swift#L926)